saleor/product/utils/product_variants.py

```diff
@@ -0,0 +1,26 @@
+from django.core.exceptions import ValidationError
+
+from ..error_codes import ProductErrorCode
+
+
+def product_variant_exist(product):
+    if product.variants.first() is None:
+        raise ValidationError(
```

Comment:
We should raise graphql error in "core utils".  If you want to raise errors in utils we should create custom exceptions like `InsufficientStock` and handle it in mutation.

Reply:
good catch :)